flashinfer-jit-cache/build_backend.py

```
@@ -82,14 +82,29 @@ def _compile_jit_cache(output_dir: Path, verbose: bool = True):
-        ["git", "submodule", "update", "--init", "--recursive"],
-        cwd=str(project_root),
-        check=True,
-    )
+    submodule_check_paths = [
+        project_root / "3rdparty" / "cutlass" / "include",
+        project_root / "3rdparty" / "spdlog" / "include",
+        project_root / "3rdparty" / "cccl" / "cub",
+    ]
+    if not all(p.exists() for p in submodule_check_paths):
+        result = subprocess.run(
+            ["git", "submodule", "update", "--init", "--recursive"],
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)